omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java

```diff
@@ -89,7 +89,7 @@
             }
         }
         else {
-            diagnosisSets = emrApiProperties.getDiagnosisSets();
+            diagnosisSets = StringUtils.isNotEmpty(diagnosisConceptSources) || StringUtils.isNotEmpty(diagnosisConceptClasses) ? null : emrApiProperties.getDiagnosisSets();
```

Comment:
We said:
>as soon as we start specifying any attribute, all defaults drop

For the sake of this controller, should the check not be
```java
diagnosisConceptSources != null || diagnosisConceptClasses != null
```
?

Reply:
Ohh, sorry I didn't clarify that actually spring places a null string/character in the requested parameter's variable. I attempted this earlier and that check failed then did a failover to apache `StringUtils`.